Setup.sh

```diff
@@ -1,144 +1,94 @@
-#!/bin/bash
+#!/usr/bin/env bash
 
-# EXPERIMENTAL!
-# This file is experimental and is not maintained directly by Improbable. Please use at your own risk.
-
-set -e -u -o pipefail
+set -e -u -o pipefail -x
 
 if [ "$(uname -s)" != "Darwin" ]; then
     echo "This script should only be used on OS X. If you are using Windows, please run Setup.bat."
     exit 1
 fi
 
-function markStartOfBlock {
-    echo "Starting: $1"
-}
-
-function markEndOfBlock {
-    echo "Finished: $1"
-}
-
 pushd "$(dirname "$0")"
 
-markStartOfBlock "$0"
+PINNED_CORE_SDK_VERSION=$(cat ./SpatialGDK/Extras/core-sdk.version)
+BUILD_DIR="$(pwd)/SpatialGDK/Build"
+CORE_SDK_DIR="${BUILD_DIR}/core_sdk"
+WORKER_SDK_DIR="$(pwd)/SpatialGDK/Source/SpatialGDK/Public/WorkerSDK"
+BINARIES_DIR="$(pwd)/SpatialGDK/Binaries/ThirdParty/Improbable"
+SCHEMA_COPY_DIR="$(pwd)/../../../spatial/schema/unreal/gdk"
+SCHEMA_STD_COPY_DIR="$(pwd)/../../../spatial/build/dependencies/schema/standard_library"
+SPATIAL_DIR="$(pwd)/../../../spatial"
+
+
+echo "Setup the git hooks"
+if [ -e .git/hooks ]; then
+    # Remove the old post-checkout hook.
+    if [ -e .git/hooks/post-checkout ]; then
+        rm -f .git/hooks/post-checkout
+    fi
 
-markStartOfBlock "Setup the git hooks"
-    if [ -e .git/hooks ]; then
-        # Remove the old post-checkout hook.
-        if [ -e .git/hooks/post-checkout ]; then rm -f .git/hooks/post-checkout; fi
+    # Remove the old post-merge hook.
+    if [ -e .git/hooks/post-merge ]; then
+        rm -f .git/hooks/post-merge
+    fi
 
-        # Remove the old post-merge hook.
-        if [ -e .git/hooks/post-merge ]; then rm -f .git/hooks/post-merge; fi
+    # Add git hook to run Setup.sh when RequireSetup file has been updated.
+    echo '#!/usr/bin/env bash'                                                              >> .git/hooks/post-merge
+    echo 'changed_files="$(git diff-tree -r --name-only --no-commit-id ORIG_HEAD HEAD)"'    >> .git/hooks/post-merge
+    echo 'check_run() {'                                                                    >> .git/hooks/post-merge
```

Comment:
Why is this a function if we just immediately invoke it below?

Reply:
This could also live in its own file and we just `cp` it across (overwriting if its already there) instead of echo'ing it line-by-line

Reply:
good point, wanted to look at it a bit more but then forgot. will do that